sdk/metric/internal/exemplar/filtered_reservoir.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exemplar // import "go.opentelemetry.io/otel/sdk/metric/internal/exemplar"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// FilteredReservoir is wraps a [Reservoir] with a filter.
+type FilteredReservoir[N int64 | float64] interface {
+	// Offer accepts the parameters associated with a measurement. The
+	// parameters will be stored as an exemplar if the filter decides to
+	// sample the measurement.
+	//
+	// The passed ctx needs to contain any baggage or span that were active
+	// when the measurement was made. This information may be used by the
+	// Reservoir in making a sampling decision.
+	Offer(ctx context.Context, val N, attr []attribute.KeyValue)
+	// Collect returns all the held exemplars in the reservoir.
+	Collect(dest *[]Exemplar)
+}
```

Comment:
Is this type needed? Can we just export the struct below? We don't ever plan to move this out of internal and there shouldn't ever be more than one implementation, right?

Reply:
I ended up adding it so that I could have Drop implement the FilteredReservoir.  If we just used `nil` instead of Drop, we could avoid needing the interface

Reply:
Gotcha. Whichever approach you think best I'm good with then.